zyngui/zynthian_gui_layer.py

```diff
@@ -507,8 +507,9 @@ def set_midi_chan_preset(self, midich, preset_index):
 					continue
 				if layer.set_preset(preset_index,True) and not selected:
 					try:
-						if not self.zyngui.modal_screen and self.zyngui.active_screen in ('control'):
-							self.select_action(self.root_layers.index(layer))
+						#following conditional statement removed to omit issue with not updating the screen
+						#if not self.zyngui.modal_screen and self.zyngui.active_screen in ('control'):
```

Comment:
We don't want preset loading to change the active screen. What you see as a bug, it's a feature ;-)
Or perhaps i'm not understanding the problem you have.

Reply:
I proposed this change to omit the issue that the screen is not updated after processing the midi bank change message and a program control change message (instrument selection). Do you have a low impact alternative solution to update the presently selected screen. For instance when the bank is updated the bank screen data shown is not changed. Same goed for the preset screen and likely also for the layer screen. This is very confusing for the user.